packages/fast-usdc/src/exos/advancer.js

```diff
@@ -9,116 +11,219 @@ import { addressTools } from '../utils/address.js';
+            const { assetManagerFacet, poolAccount, usdcBrand } = this.state;
+            const { recipientAddress } = evidence.aux;
+            const { EUD } =
+              addressTools.getQueryParams(recipientAddress).params;
+            if (!EUD) {
+              throw makeError(
+                `recipientAddress does not contain EUD param: ${q(recipientAddress)}`,
+              );
+            }
 
-        // TODO #10391 this can throw, and should make a status update in the catch
-        const destination = chainHub.makeChainAddress(EUD);
+            // this will throw if the bech32 prefix is not found, but is handled by the catch
+            const destination = chainHub.makeChainAddress(EUD);
 
-        /** @type {DenomAmount} */
-        const requestedAmount = harden({
-          denom: this.state.localDenom,
-          value: BigInt(evidence.tx.amount),
-        });
+            const requestedValue = BigInt(evidence.tx.amount);
+            const requestedAmount = AmountMath.make(usdcBrand, requestedValue);
+            const poolBalance = assetManagerFacet.lookupBalance();
 
-        // TODO #10391 ensure there's enough funds in poolAccount
+            if (!AmountMath.isGTE(poolBalance, requestedAmount)) {
+              log(
+                `Insufficient pool funds`,
+                `Requested ${q(requestedAmount)} but only have ${q(poolBalance)}`,
+              );
+              statusManager.observe(evidence);
+              return;
+            }
 
-        const transferV = E(this.state.poolAccount).transfer(
-          destination,
-          requestedAmount,
-        );
+            try {
+              // mark as Advanced since `transferV` initiates the advance
+              // will throw if we've already .skipped or .advanced this evidence
+              statusManager.advance(evidence);
+            } catch (e) {
+              // only anticipated error is `assertNotSeen`, so
+              // intercept the catch so we don't call .skip which
+              // also performs this check
+              log('Advancer error:', q(e).toString());
+              return;
+            }
 
-        // mark as Advanced since `transferV` initiates the advance
-        statusManager.advance(evidence);
+            try {
+              // should LiquidityPool return a vow here?
+              const { USDC: advancePmtP } =
+                await assetManagerFacet.borrowUnderlying(requestedAmount);
 
-        return watch(transferV, transferHandler, {
-          destination,
-          amount: requestedAmount.value,
-        });
+              // do we actually need to await here?
+              const advancePmt = await advancePmtP;
+              const depositV = E(poolAccount).deposit(advancePmt);
+              return watch(depositV, this.facets.depositHandler, destination);
+            } catch (e) {
+              // TODO how should we think about failure here?
```

Comment:
the deposit is unlikely to fail but the borrow could if the balance changes since it was looked up. Even though we're using a ZCFSeat we should design for if this were an ICA.

Reply:
Ok I've put a TODO for `borrow` failing. I think we can flesh that out more when we integrate the with the `LiquidityPool`.

For `.deposit()` failing, we can catch this separately in `onRejected()` and attempt to return the payment. The LP might need an interface for this, or `return()` will need to be trained to handle this scenario (right now, it's expected fees + principal).